tap_google_sheets/transform.py

```diff
@@ -0,0 +1,231 @@
+        return transform_sheet_datetime_data(value, sheet_title, col_name, col_letter, row_num, col_type)
+
+    # DATE
+    elif col_type == 'numberType.DATE':
+        return transform_sheet_date_data(value, sheet_title, col_name, col_letter, row_num, col_type)
+
+    # TIME ONLY (NO DATE)
+    elif col_type == 'numberType.TIME':
+        return transform_sheet_time_data(value, sheet_title, col_name, col_letter, row_num, col_type)
+
+    # NUMBER (INTEGER AND FLOAT)
+    elif col_type == 'numberType':
+        return transform_sheet_number_data(value, sheet_title, col_name, col_letter, row_num, col_type)
+
+    # STRING
+    elif col_type == 'stringValue':
+        return str(value)
+
+    # BOOLEAN
+    elif col_type == 'boolValue':
+        return transform_sheet_boolean_data(value, sheet_title, col_name, col_letter, row_num, col_type, row)
+
+    # OTHER: Convert everything else to a string
+    else:
+        LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR; SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+            sheet_title, col_name, col_letter, row, col_type))
+        return str(value)
+
+# Transform sheet_data: add spreadsheet_id, sheet_id, and row, convert dates/times
+#  Convert from array of values to JSON with column names as keys
+def transform_sheet_data(spreadsheet_id, sheet_id, sheet_title, from_row, columns, sheet_data_rows):
+    sheet_data_tf = []
+    row_num = from_row
+    # Create sorted list of columns based on columnIndex
+    cols = sorted(columns, key=lambda i: i['columnIndex'])
+
+    # LOGGER.info('sheet_data_rows: {}'.format(sheet_data_rows))
+    for row in sheet_data_rows:
+        # If empty row, SKIP
+        if row == []:
+            LOGGER.info('EMPTY ROW: {}, SKIPPING'.format(row_num))
+        else:
+            sheet_data_row_tf = {}
+            # Add spreadsheet_id, sheet_id, and row
+            sheet_data_row_tf['__sdc_spreadsheet_id'] = spreadsheet_id
+            sheet_data_row_tf['__sdc_sheet_id'] = sheet_id
+            sheet_data_row_tf['__sdc_row'] = row_num
+            col_num = 1
+            for value in row:
+                # Select column metadata based on column index
+                col = cols[col_num - 1]
+                col_skipped = col.get('columnSkipped')
+                if not col_skipped:
+                    # Get column metadata
+                    col_name = col.get('columnName')
+                    col_type = col.get('columnType')
+                    col_letter = col.get('columnLetter')
+
+                    # get column value based on the type of the value
+                    col_val = get_column_value(value, sheet_title, col_name, col_letter, row_num, col_type, row)
+
+                    sheet_data_row_tf[col_name] = col_val
+                col_num = col_num + 1
+            # APPEND non-empty row
+            sheet_data_tf.append(sheet_data_row_tf)
+        row_num = row_num + 1
+    return sheet_data_tf, row_num
```

Comment:
Did you guys pylint on this transform function? Some of the issues needs to be fixed for readability

Reply:
Pylint is running on **tap_google_sheets** folder